File: 1287-element-appearing-more-than-25-in-sorted-array/1287-element-appearing-more-than-25-in-sorted-array.cpp
class Solution {
public:
    int findSpecialInteger(vector<int>& arr) {
        unordered_map<int, int> cnt;
        int target = arr.size()/4;
        for(int x : arr){
            cnt[x]++;
            if(cnt[x] > target) return x;
        }
        return -1;
    }
};

File: 1287-element-appearing-more-than-25-in-sorted-array/1287-element-appearing-more-than-25-in-sorted-array.java
class Solution {
    public int findSpecialInteger(int[] arr) {
        Map<Integer, Integer> cnt = new HashMap();
        int target = arr.length/4;
        for(int x : arr){
            cnt.put(x, cnt.getOrDefault(x, 0)+1);
            if(cnt.get(x) > target) return x;
        }
        return -1;
    }
}